app/components/Views/TransactionsView/index.js

```diff
@@ -220,16 +218,60 @@ TransactionsView.propTypes = {
   chainId: PropTypes.string,
 };
 
-const mapStateToProps = (state) => ({
-  conversionRate: selectConversionRate(state),
-  currentCurrency: selectCurrentCurrency(state),
-  tokens: selectTokens(state),
-  selectedAddress: selectSelectedAddress(state),
-  identities: selectIdentities(state),
-  transactions: state.engine.backgroundState.TransactionController.transactions,
-  networkType: selectProviderType(state),
-  chainId: selectChainId(state),
-});
+const mapStateToProps = (state) => {
+  const selectedAddress = selectSelectedAddress(state);
+  const chainId = selectChainId(state);
+  const nonSmartTransactions =
+    state.engine.backgroundState.TransactionController.transactions; // these are transactionMeta objs
+  const smartTransactions =
+    state.engine.backgroundState.SmartTransactionsController
+      ?.smartTransactionsState?.smartTransactions?.[chainId] || [];
+
+  // Remove duplicate confirmed STX
+
+  // for replaced txs, only hide the ones that are confirmed
+  const filteredNonSmartTransactions = nonSmartTransactions.filter(
+    (tx) => !(tx.replacedBy && tx.replacedById && tx.hash),
+  );
+
+  const filteredPendingSmartTransactions =
+    smartTransactions
+      ?.filter((stx) => {
+        const { txParams } = stx;
+        return (
+          txParams?.from.toLowerCase() === selectedAddress.toLowerCase() &&
+          stx.status &&
+          stx.status !== SmartTransactionStatuses.SUCCESS &&
+          stx.status !== SmartTransactionStatuses.CANCELLED
```

Comment:
The first check is redundant.
```suggestion
          stx.status !== SmartTransactionStatuses.SUCCESS &&
          stx.status !== SmartTransactionStatuses.CANCELLED
```
or even
```suggestion
          ![SmartTransactionStatuses.SUCCESS, SmartTransactionStatuses.CANCELLED].includes(stx.status)
```